java/src/main/java/com/logicalclocks/base/TrainingDatasetBase.java

```diff
@@ -0,0 +1,412 @@
+
+  /**
+   * Update the statistics configuration of the training dataset.
+   * Change the `enabled`, `histograms`, `correlations` or `columns` attributes and persist
+   * the changes by calling this method.
+   *
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  public abstract void updateStatisticsConfig() throws FeatureStoreException, IOException;
+
+  /**
+   * Get the last statistics commit for the training dataset.
+   *
+   * @return statistics object of latest commit
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  @JsonIgnore
+  public abstract Statistics getStatistics() throws FeatureStoreException, IOException;
+
+  /**
+   * Get the statistics of a specific commit time for the training dataset.
+   *
+   * @param commitTime commit time in the format "YYYYMMDDhhmmss"
+   * @return statistics object for the commit time
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  @JsonIgnore
+  public abstract Statistics getStatistics(String commitTime) throws FeatureStoreException, IOException;
+
+  /**
+   * Add name/value tag to the training dataset.
+   *
+   * @param name  name of the tag
+   * @param value value of the tag. The value of a tag can be any valid json - primitives, arrays or json objects
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  public abstract void addTag(String name, Object value) throws FeatureStoreException, IOException;
+
+  /**
+   * Get all tags of the training dataset.
+   *
+   * @return a map of tag name and values. The value of a tag can be any valid json - primitives, arrays or json objects
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  @JsonIgnore
+  public abstract Map<String, Object> getTags() throws FeatureStoreException, IOException;
+
+  /**
+   * Get a single tag value of the training dataset.
+   *
+   * @param name name of the tag
+   * @return The value of a tag can be any valid json - primitives, arrays or json objects
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  @JsonIgnore
+  public abstract Object getTag(String name) throws FeatureStoreException, IOException;
+
+  /**
+   * Delete a tag of the training dataset.
+   *
+   * @param name name of the tag to be deleted
+   * @throws FeatureStoreException
+   * @throws IOException
+   */
+  public abstract void deleteTag(String name) throws FeatureStoreException, IOException;
+
+  /**
+   * Delete training dataset and all associated metadata.
+   * Note that this operation drops only files which were materialized in
+   * HopsFS. If you used a Storage Connector for a cloud storage such as S3,
+   * the data will not be deleted, but you will not be able to track it anymore
+   * from the Feature Store.
+   * This operation drops all metadata associated with this version of the
+   * training dataset and and the materialized data in HopsFS.
```

Comment:
```suggestion
   * training dataset and the materialized data in HopsFS.
```
The same issues also appears in the `TrainingDataset` class. This also leads me to wonder, if maybe the Docstring duplication should be avoided and it should be stored only in the base class.